app/controllers/donations_controller.rb

```diff
@@ -11,6 +11,15 @@ def create
     return_to_origin_page
   end
 
+  def undo
+    params[:donation] = {
+                         "candidate_issue_id" => session[:undo].keys.first,
```

Comment:
Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.